Add full CRD support back into KCP controlplane

davidfestal opened the pull request:
Add full CRD support back into the KCP control plane.
In the current context of what Kubernetes is, the most straightforward way to do it is to
bring back the aggregation apiServer (even if it will not be used to delegate to external
aggregated servers) since it contains part of the logic that initializes and leverages the
CRD apiExtensions server.

davidfestal commented:
cc @smarterclayton

smarterclayton commented on pkg/controlplane/aggregator.go:
```diff
@@ -0,0 +1,326 @@
+	{Group: "autoscaling", Version: "v2beta1"}:                   {group: 17500, version: 9},
+	{Group: "autoscaling", Version: "v2beta2"}:                   {group: 17500, version: 1},
+	{Group: "batch", Version: "v1"}:                              {group: 17400, version: 15},
+	{Group: "batch", Version: "v1beta1"}:                         {group: 17400, version: 9},
```
These priorities are hardcoded... so they need to be removed

smarterclayton replied:
You probably want a HACK commit that removes all the ones you don't care about and makes it more precise, but the point of the generic code was that we shouldn't need to know about these.  Having one commit that reintroduces the file, and one commit that removes all the "doesn't belong in a generic apiserver"  is probably the easiest way to remember this when we eventually have to rebase and get it into shape for an upstream change

davidfestal replied:
Yes, I'll add a commit to remove the lines we don't need because they "don't belong in a generic apiserver".

davidfestal replied:
Done in commit https://github.com/smarterclayton/kubernetes/pull/6/commits/22052930192b135103131eab1ecf175fb6cac612

davidfestal committed:
GENERIC: Drop groups that don't belong in a generic apiserver

Signed-off-by: David Festal <dfestal@redhat.com>

davidfestal committed:
RESTORE: Bring back the Aggregator...

... since it is required to get back
fully functional CRDs inside the control plane
minimal generic apiserver.

Signed-off-by: David Festal <dfestal@redhat.com>

davidfestal committed:
NEW: Now use the aggregator ...

... in the control plane `CreateServerChain`
function.

Signed-off-by: David Festal <dfestal@redhat.com>
